example/12_reduce/reduce_blockwise_two_call.cpp

```diff
@@ -0,0 +1,291 @@
+                                                     InOutDataType,
+                                                     4, // Rank
+                                                     1, // NumReduceDim
+                                                     ReduceOperation,
+                                                     PassThroughOp,
+                                                     AccElementwiseOperation,
+                                                     PropagateNan,
+                                                     OutputIndex,
+                                                     false, // HaveIndexInputIfOutputIndex
+                                                     256,
+                                                     256,
+                                                     1,
+                                                     1,
+                                                     1,
+                                                     1, // vector dim
+                                                     1,
+                                                     1>;
+
+static bool do_verify;
+static int init_method;
+static float alpha;
+static float beta;
+static bool time_kernel;
+
+int main(int argc, char* argv[])
+{
+    // used by the device reduction
+    const std::vector<int> reduceDims_1    = {4};
+    const std::vector<int> invariantDims_1 = {0, 1, 2, 3};
+
+    const std::vector<int> reduceDims_2    = {3};
+    const std::vector<int> invariantDims_2 = {0, 1, 2};
+
+    // used by the host reduction
+    const std::vector<int> reduceDims    = {3, 4};
+    const std::vector<int> invariantDims = {0, 1, 2};
+
+    const std::vector<size_t> inLengths_1 = {64, 320, 80, 4, 128};
+
+    using namespace ck::host_reduce;
+    using ck::host_common::to_int_vector;
+
+    if(argc == 1)
+    {
+        do_verify   = true;
+        init_method = 2;
+        time_kernel = true;
+    }
+    else if(argc == 4)
+    {
+        do_verify   = static_cast<bool>(argv[1]);
+        init_method = atoi(argv[2]);
+        time_kernel = static_cast<bool>(atoi(argv[3]));
+    }
+    else
+    {
+        std::ostringstream ostr;
+
+        ostr << "Wrong parameter! " << std::endl
+             << "Usage: " << argv[0] << "[verify 0/1] init_method time_kernel" << std::endl;
+
+        throw std::runtime_error(ostr.str());
+    };
+
+    alpha = 1.0f;
+    beta  = 0.0f;
+
+    Tensor<InOutDataType> in_1(inLengths_1);
+
+    std::vector<size_t> inLengths_2; // also the output lengths of the first reduction
+    std::vector<size_t> outLengths;
+
+    if(invariantDims_1.empty())
+        inLengths_2.push_back(1);
+    else
+        for(auto dim : invariantDims_1)
+            inLengths_2.push_back(inLengths_1[dim]);
+
+    if(invariantDims_2.empty())
+        outLengths.push_back(1);
+    else
+        for(auto dim : invariantDims_2)
+            outLengths.push_back(inLengths_2[dim]);
+
```

Comment:
I think it's better to specify inLengths_2 and inStrides_2 directly, something like:
```
const std::vector<size_t> inLengths_2 = {xx, xx, xxx, xxx};
const std::vector<size_t> inStrides_2 = {xx, xx, xxx, xxx};
```
Otherwise reader of this example need to decihper your code to understand what you are doing. The purpose of this example is for read to understand what we can do with our code.

Reply:
Good suggestion!

Reply:
Corrected by  [1c4ce](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/237/commits/1d1435b40c1088562d193bb462d33cea7ac1c4ce)